guides/Codable.md

```diff
@@ -0,0 +1,102 @@
+## I want to make my types conform to `Encodable` or `Decodable`
+
+### [Encodable Template](https://github.com/krzysztofzablocki/Sourcery/blob/master/Templates/Templates/Encodable.stencil)
+### [Decodable Template](https://github.com/krzysztofzablocki/Sourcery/blob/master/Templates/Templates/Encodable.stencil)
+
+#### Available annotations
+
+
+* `skipCoding` - skips decoding and encoding
+* `codingKey` - The coding key to use, this is optional and if no `codingKey` annotation is found the variable name is used.
+* `codingDefaultValue` - what to use if nil on non optional variable, defaults to 0/""/etc
+
+**Example Input**
+
+```swift
+protocol AutoEncodable: Codable { }
+
+protocol AutoDecodable: Decodable { }
+
+protocol AutoCodable: AutoEncodable, AutoDecodable { }
+
+struct AutoDecodableTest: AutoCodable {
+  // sourcery: codingKey="auto_codable_bool", encodingType="Int"
+  let bool: Bool
+  let optional: String?
+  // sourcery: codingKey="auto_codable_prim_1"
+  let prim1: Int
+  // sourcery: codingKey="auto_codable_prim_2"
+  let prim2: String
+  // sourcery: codingKey="auto_codable_prim_3"
+  let prim3: Double
+  let arr: [String]
+  let arrOpt: [String]?
+  // sourcery: codingDefaultValue=Date()
+  let date: Date
+  let optDate: Date?
+}
+```
+
+**Example Output**
+
+```swift
+// file = AutoDecodable.generated.swift
+
+// MARK: AutoDecodableTest+Decodable | AutoDecodable
+extension AutoDecodableTest {
+  private enum CodableKeys: String, CodingKey {
+    case bool  = "auto_codable_bool"
+    case optional 
+    case prim1  = "auto_codable_prim_1"
+    case prim2  = "auto_codable_prim_2"
+    case prim3  = "auto_codable_prim_3"
+    case arr 
+    case arrOpt 
+    case date 
+    case optDate 
+  }
+
+  internal init(from decoder: Decoder) throws {
```

Comment:
I discovered that Swift 4.1 will warn that this constructor does not call member-wise constructor, because of cross-module constructors changes. So we need to take care of that in the template, which will be harder as there might be no such constructor (maybe we should generate one if we can't find it in type)